File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/AcidSlime_S_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class AcidSlime_S_Model extends MonsterInfo{
    public static Id = "AcidSlime_S"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 2)
            return Mathf.RandomRange(9, 13);
        else
            return Mathf.RandomRange(8,12);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/slimeS/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/slimeS/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/slimeS/skeleton.png"));
    }

    get Name(): string {
        return "酸液史莱姆（小）";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/AcidSlime_S_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class AcidSlime_S_Model : MonsterInfo
    {
        public static string Id = "AcidSlime_S";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(9, 13) : Random.Range(8, 12);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/slimeS/skeleton.atlas",
                    "Monsters/theBottom/slimeS/skeleton.json", "Monsters/theBottom/slimeS/skeleton.png");
            }
        }

        public override string Name => "酸液史莱姆（小）";

        public AcidSlime_S_Model(Vector2 pos) : base(pos)
        {
        }
    }
}